src/gc/CMakeLists.txt

```diff
@@ -26,12 +16,48 @@ set( GC_SOURCES
   gceewks.cpp
   handletablecache.cpp)
 
-if(NOT CLR_CMAKE_PLATFORM_UNIX)
-set ( GC_SOURCES
-  ${GC_SOURCES}
-  windows/gcenv.windows.cpp)
-endif(NOT CLR_CMAKE_PLATFORM_UNIX)
+if(CLR_CMAKE_PLATFORM_UNIX)
+  include(unix/configure.cmake)
+  set ( GC_SOURCES
+    ${GC_SOURCES}
+    unix/gcenv.unix.cpp
+    unix/events.cpp
+    unix/cgroup.cpp)
+else()
+  set ( GC_SOURCES
+    ${GC_SOURCES}
+    windows/gcenv.windows.cpp)
+endif(CLR_CMAKE_PLATFORM_UNIX)
+
+if(WIN32)
+  set (GC_LINK_LIBRARIES
+    ${STATIC_MT_CRT_LIB}
+    ${STATIC_MT_VCRT_LIB}
+    kernel32.lib)
+else()
+  set (GC_LINK_LIBRARIES)
+endif(WIN32)
 
 convert_to_absolute_path(GC_SOURCES ${GC_SOURCES})
 
-add_library_clr(gc_standalone STATIC ${GC_SOURCES})
+add_library(gc SHARED ${GC_SOURCES})
+target_link_libraries(gc ${GC_LINK_LIBRARIES})
+install_clr(gc)
+
+if(CLR_CMAKE_PLATFORM_UNIX)
+  target_compile_options(gc PUBLIC -fPIC)
+  # dprintf causes many warnings
+  target_compile_options(gc PUBLIC -Wno-format)
+endif(CLR_CMAKE_PLATFORM_UNIX)
+
+target_compile_definitions(gc PUBLIC -DBUILD_AS_STANDALONE)
+if(CLR_CMAKE_PLATFORM_DARWIN)
+  # The implementation of GCToOSInterface on MacOS makes use of non-POSIX
+  # pthreads APIs, which by default are not included in the pthreads header
+  # unless we define this macro.
+  target_compile_definitions(gc PUBLIC -D_DARWIN_C_SOURCE)
+endif(CLR_CMAKE_PLATFORM_DARWIN)
+
+target_include_directories(gc BEFORE PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(gc BEFORE PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/env)
```

Comment:
Ah, seems I was misunderstanding the scope of `add_compile_options` and `include_directories`. Will fix.